### ISSUE TITLE
Enhance Ditto search to be able to query for equality of JsonObjects + JsonArrays

### ISSUE THREAD

**Issue by thjaeckle** [eclipse-ditto/ditto#1804]
Currently, it is only possible to query e.g. for `eq` using scalar Json values, like Json String, Number, Boolean.

It would also be very useful to e.g. query for equality of full Json Objects, e.g. like:
```
eq(attributes/location,{"longitude":23.12313,"latitude":52.23123})
```

Predicates which could support this:
* `eq`
* `ne`
* `in` ?

**Comment by Abhijeetmishr** [comment]
Hi @thjaeckle can I work on this ?

**Comment by thjaeckle** [comment]
@Abhijeetmishr sure, that would be great. You already collected some experience enhancing the RQL with `ilike`, right?

**Comment by Abhijeetmishr** [comment]
> @Abhijeetmishr sure, that would be great. You already collected some experience enhancing the RQL with `ilike`, right?

@thjaeckle Yes Thomas you are right.
I will be sharing my approach soon with you.

**Comment by Abhijeetmishr** [comment]
Hi @thjaeckle Thomas, 
Basically we wanted all the predicates to accept JsonObject and Arrays inplace of JsonValue.
![image](https://github.com/eclipse-ditto/ditto/assets/43075376/146a1f0b-b632-4761-9fb2-167e63ec648f)

But I wanted to understand the whole code flow before jumping to the implementation part how can I test it ?

**Comment by thjaeckle** [comment]
@Abhijeetmishr no, not all predicates .. as mentioned in the description this only makes sense for:
> Predicates which could support this:
> * `eq`
> * `ne`
> * `in` ?

You cannot compare if a JsonObject is "greater than" another JsonObject in a meaningful way.

How to test?
E.g. via `ThingPredicatePredicateVisitorTest` for the Java based predicate evaluation.
For the Mongo BSON based ones, the `SearchActorIT` is probably the unit test of choice to e.g. enhance then..
